20230311/mike.java

```diff
@@ -0,0 +1,34 @@
+/*
+Use slow and fast pointer to find mid node(just like recent questions).
+If we find mid node, cut it into two part as left subtree and right subtree.
+*/
+class Solution {
+    public TreeNode sortedListToBST(ListNode head) {
+        if(head==null) {
+            return null;
+        }
+        
+        return findRootNodeInMid(head,null);
+    }
+    
+    public TreeNode findRootNodeInMid(ListNode head, ListNode tail){
+        ListNode slow = head;
+        ListNode fast = head;
```

Comment:
oh yeah! using fast-slow method to get the middle node is more efficient. well done.